src/osd/scrubber/ScrubStore.cc

```diff
@@ -150,21 +160,30 @@ void Store::add_snap_error(int64_t pool, const inconsistent_snapset_wrapper& e)
 
 bool Store::empty() const
 {
-  return results.empty();
+  return results.empty() && deep_results.empty();
 }
 
 void Store::flush(ObjectStore::Transaction* t)
 {
   if (t) {
+
     OSDriver::OSTransaction txn = driver.get_transaction(t);
     backend.set_keys(results, &txn);
+
+    OSDriver::OSTransaction deep_txn = deep_driver.get_transaction(t);
+    deep_backend.set_keys(deep_results, &deep_txn);
+
   }
   results.clear();
+  deep_results.clear();
 }
 
-void Store::cleanup(ObjectStore::Transaction* t)
+void Store::cleanup(ObjectStore::Transaction* t, scrub_level_t level)
```

Comment:
It's not mandated by the code guidelines but I personally prefer to qualify as `const` to blow compilation out in case of stupid typo like `=` instead of `==`.